namespaces/mson-namespace.md

```diff
@@ -390,29 +471,35 @@ Using the `ref` element to reference an the content of an element.
 }
 ```
 
-Using `extend` to merge objects together.
+Using "Type Reference" (`ref`) element with the `resolved` attribute:
 
 ```json
 {
-    "element": "extend",
+    "element": "object",
     "content": [
         {
-            "element": "object",
-            "content": [
-                {
-                    "element": "member",
-                    "content": {
-                        "key": {
-                            "element": "string",
-                            "content": "id"
-                        }
-                    }
+            "element": "member",
+            "content": {
+                "key": {
+                    "element": "string",
+                    "content": "id"
                 }
-            ]
+            }
         },
         {
             "element": "ref",
-            "content": "User"
+            "attributes": {
+                "resolved": {
+                    "element": "object",
+                    "meta": {
+                        "ref": "User"
```

Comment:
> I think we should define what the issues are we're seeing. We may be on a slippery slope because we're not doing this correctly.

I will think about it. The whole inheritance feels complex but I wasn't able to come with a simpler data based solution.